pulp_container/app/models.py

```diff
@@ -173,9 +173,8 @@ class Tag(Content):
 
     name = models.CharField(max_length=255, db_index=True)
 
-    # TODO This should be 'null=False'.
     tagged_manifest = models.ForeignKey(
-        Manifest, null=True, related_name="tagged_manifests", on_delete=models.CASCADE
+        Manifest, null=False, related_name="tagged_manifests", on_delete=models.CASCADE
```

Comment:
```suggestion
        Manifest, null=False, related_name="tagged_manifests", on_delete=models.CASCADE
```
Should we also PROTECT from removal? (Not quite sure if it'd break orphan cleanup...)

Reply:
I think CASCADE is the right to do here, tag is removed with the manifest since tags can't exit without the manifest.

Reply:
That is why PROTECT would protect the manifest not to be deleted as long as a tag is pointing there.

Reply:
It is a bad idea, https://github.com/pulp/pulpcore/blob/main/pulpcore/app/tasks/orphan.py#L66 if first a manifest will be processed then protection error will be raised.
What is the value to protect tagged manifest if it is already orphaned?
I don't think in pulp we have already a case where one content is protected by the other.